xmtp_id/src/scw_verifier.rs

```diff
@@ -56,6 +66,69 @@ impl ERC1271Verifier {
 
         Ok(res == EIP1271_MAGIC_VALUE)
     }
+
+    /// Verifies an ERC-6492<https://eips.ethereum.org/EIPS/eip-6492> signature.
+    ///
+    /// # Arguments
+    ///
+    /// * `block_number` - Block number to verify the signature at.
+    /// * `signer` - can be the smart wallet address or the ecrecover .
+    /// * `hash` - Message digest for the signature.
+    /// * `signature` - Could be encoded smart wallet signature or raw ECDSA signature.
+    pub async fn erc6492_is_valid_signature(
+        &self,
+        block_number: Option<BlockNumber>,
+        signer: Address,
+        hash: [u8; 32],
+        signature: Bytes,
+    ) -> Result<bool, VerifierError> {
+        let code = hex::decode(VALIDATRE_SIG_OFF_CHAIN_BYTECODE)?;
+        // ABI of the ValidateSigOffchain constructor
+        // constructor (address _signer, bytes32 _hash, bytes memory _signature)
+        let inputs: Vec<Param> = vec![
+            Param {
+                name: "_signer".to_string(),
+                kind: ParamType::Address,
+                internal_type: Some("_signer".to_string()),
+            },
+            Param {
+                name: "_hash".to_string(),
+                kind: ParamType::Bytes,
+                internal_type: Some("_hash".to_string()),
+            },
+            Param {
+                name: "_signature".to_string(),
+                kind: ParamType::Bytes,
+                internal_type: Some("_signature".to_string()),
+            },
+        ];
+        let constructor = Constructor { inputs };
+
+        let tokens = &[
+            Token::Address(signer),
+            Token::Bytes(hash.to_vec()),
+            Token::Bytes(signature.to_vec()),
+        ];
+        let data = constructor.encode_input(code, tokens)?;
+
+        let tx: TypedTransaction = TransactionRequest::new()
+            .from(
+                // Since this is a eth_call, we need to make sure the `from` address has balance to complete the simulation
```

Comment:
Very useful comment.

The signature validator code doesn't take `message.sender` into account anywhere?

Reply:
Nope, the `message.sender` deploys the validation contract with some arguments for the actual validation.